src/main/java/bookstore/model/RoleName.java

```diff
@@ -0,0 +1,6 @@
+package bookstore.model;
+
+public enum RoleName {
+    USER,
+    ADMIN
```

Comment:
as far as I remember `hasRole` automatically adds `ROLE_` prefix to the arguments

it means that `hasRole("USER")` with check for `ROLE_USER` role - but here you have just `USER`

pls double check whether everything works with current setup and let me know in the chat :)

Reply:
I did some refactoring and now it works